lib/ffmpeg/transcoder.rb

```diff
@@ -145,5 +138,42 @@ def fix_encoding(output)
     rescue ArgumentError
       output.force_encoding("ISO-8859-1")
     end
+
+    def optimize_screenshot_parameters(options, transcoder_options)
```

Comment:
(if you're interested in feedback): I'm worried that this method is doing too many things at the same time while having confusing name. It doesn't only optimize screenshot parameters but it also handles wrapping different ways of passing arguments to it. Might be a good idea to split it into several smaller methods, even if for the sake of readability.